File: 509. Fibonacci Number.py

#recursive
class Solution:
    def fib(self, N):

        if N == 0:
            return 0
        elif N == 1:
            return 1
        else:
            return self.fib(N - 1) + self.fib(N - 2)

        
#iterative        
class Solution:
    def fib(self, N):

        if N == 0:
            return 0
        elif N == 1:
            return 1
        pre1 = 1
        pre2 = 0
        for i in range(2, N+1):
            res = pre1 + pre2
            pre2 = pre1
            pre1 = res
        return res

File: Array/509. Fibonacci Number.c

#recursive
int fib(int N){
    if(N == 0){
        return 0;
    }
    else if(N == 1){
        return 1;
    }
    else{
        return fib(N - 1) + fib(N - 2);
    }

}




#iterative
int fib(int N){
    if(N == 0){
        return 0;
    }
    else if(N == 1){
        return 1;
    }
    int pre1 = 1;
    int pre2 = 0;
    int res = 0;
    for(int i = 2; i <= N; i++){
        res = pre1 + pre2;
        pre2 = pre1;
        pre1 = res;
    }return res;
}
